packages/cli/internal/pkg/aws/cwl/get_logs.go

```diff
@@ -51,9 +51,14 @@ func (c Client) GetLogsPaginated(input GetLogsInput) LogPaginator {
 }
 
 func formatEvents(events []types.FilteredLogEvent) []string {
-	logs := make([]string, len(events))
-	for i, event := range events {
-		logs[i] = formatEvent(event)
+	logsByStream := make(map[string][]string)
+	for _, event := range events {
+		logsByStream[*event.LogStreamName] = append(logsByStream[*event.LogStreamName], formatEvent(event))
+	}
+
+	var logs []string
```

Comment:
I misread the AWS documentation as I thought that the response size was smaller than it actually is. I'll refactor it so that we use the same memory footprint

Reply:
The new change now will only force a max memory increase of 640kb over the existing implementation of AGC and the current response from AWS max memory usage would be up to 2.56Gb

Calculations:
Max Memory usage - 64 bytes * 10 000 events = 640 kilobytes
Max Cloudwatch memory usage - 256 kb * 10 000 events = 2.56Gb